lib/metasploit/framework/login_scanner/mybook_live.rb

```diff
@@ -33,17 +33,16 @@ def attempt_login(credential)
             result_opts[:service_name] = 'http'
           end
           begin
-            cred = Rex::Text.uri_encode(credential.private)
-            body = "data%5BLogin%5D%5Bowner_name%5D=admin&data%5BLogin%5D%5Bowner_passwd%5D=#{cred}"
-            cli = Rex::Proto::Http::Client.new(host, port, {'Msf' => framework, 'MsfExploit' => framework_module}, ssl, ssl_version, http_username, http_password)
-            configure_http_client(cli)
-            cli.connect
-            req = cli.request_cgi(
+            cred = Rex::Text.uri_encode(credential.private) 
+            res = send_request({
               'method' => method,
               'uri' => uri,
-              'data' => body
-            )
-            res = cli.send_recv(req)
+              'vars_post' => {
+                 'data%5BLogin%5D%5Bowner_name%5D' => 'admin',
```

Comment:
Place holder comment; we might need to double check that this doesn't get double encoded, i.e. `%` becoming `%25`

So this _might_ need to be

```
                 'data[Login][owner_name]' => 'admin',
                 'data[Login][owner_passwd]' => cred

```

Reply:
Running the module against a local ncat server:

```
msf6 auxiliary(scanner/http/mybook_live_login) >  run http://127.0.0.1:4444 password=password123
```

On master:

```
ncat -klnvp 4444
Ncat: Version 7.93 ( https://nmap.org/ncat )
Ncat: Listening on :::4444
Ncat: Listening on 0.0.0.0:4444
Ncat: Connection from 127.0.0.1.
Ncat: Connection from 127.0.0.1:49407.
Ncat: Connection from 127.0.0.1.
Ncat: Connection from 127.0.0.1:49408.
POST /UI/login HTTP/1.1
Host: 127.0.0.1:4444
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 13.1; rv:108.0) Gecko/20100101 Firefox/108.0
Content-Type: application/x-www-form-urlencoded
Content-Length: 83

data%5BLogin%5D%5Bowner_name%5D=admin&data%5BLogin%5D%5Bowner_passwd%5D=password123
```

On this branch no body is sent:
```
ncat -klnvp 4444
Ncat: Version 7.93 ( https://nmap.org/ncat )
Ncat: Listening on :::4444
Ncat: Listening on 0.0.0.0:4444
Ncat: Connection from 127.0.0.1.
Ncat: Connection from 127.0.0.1:49428.
Ncat: Connection from 127.0.0.1.
Ncat: Connection from 127.0.0.1:49436.
POST /UI/login HTTP/1.1
Host: 127.0.0.1:4444
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 13_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36
Content-Length: 0


```

Reply:
If we add `cgi: true` to the `send_request` call, the body will be sent through now:

```
msf6 auxiliary(scanner/http/mybook_live_login) > run http://127.0.0.1:4444 password=password123 httptrace=true

From: /Users/adfoster/Documents/code/metasploit-framework/lib/metasploit/framework/login_scanner/mybook_live.rb:38 Metasploit::Framework::LoginScanner::MyBookLive#attempt_login:

    33:             result_opts[:service_name] = 'http'
    34:           end
    35:           begin
    36:             cred = Rex::Text.uri_encode(credential.private)
    37:             require 'pry-byebug'; binding.pry
 => 38:             res = send_request({
    39:               'method' => method,
    40:               'uri' => uri,
    41:               cgi: true,
    42:               'vars_post' => {
    43:                  'data%5BLogin%5D%5Bowner_name%5D' => 'admin',

[1] pry(#<Metasploit::Framework::LoginScanner::MyBookLive>)> continue

####################
# Request:
####################
POST /UI/login HTTP/1.1
Host: 127.0.0.1:4444
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 13.1; rv:108.0) Gecko/20100101 Firefox/108.0
Content-Type: application/x-www-form-urlencoded
Content-Length: 99

data%255BLogin%255D%255Bowner_name%255D=admin&data%255BLogin%255D%255Bowner_passwd%255D=password123
```

And it gets double encoded as I expected; So looks like this needs to be updated to set cgi to true, as well as removing extra level of entity encoding :+1: